osv/purl_helpers.py

```diff
@@ -63,7 +107,8 @@ def package_to_purl(ecosystem: str, package_name: str) -> str | None:
   return f'pkg:{purl_type}/{_url_encode(package_name)}{suffix}'
 
 
-def purl_to_ecosystem(purl_type: str) -> str | None:
-  """Convert purl to a specific ecosystem string"""
-  ecosystem_purl = {v: k for k, v in PURL_ECOSYSTEMS.items()}
-  return ecosystem_purl.get(purl_type)
+def parse_purl_ecosystem(purl: PackageURL) -> str | None:
```

Comment:
Maybe this one function can parse out the ecosystem, name and version, just so we don't have to extract it ourselves later.